src/Core/PackageKitBackend.vala

```diff
@@ -42,6 +42,26 @@ public class AppCenterCore.PackageKitBackend : Backend, Object {
 
     public bool working { get; private set; }
 
+    // The aptcc backend included in PackageKit < 1.1.10 wasn't able to support multiple packages
+    // passed to the search_names method at once. If we have a new enough version we can enable
+    // some optimisations when looking up packages
+    public bool supports_parallel_package_queries {
+        get {
+            var control = new Pk.Control ();
```

Comment:
Because a `Pk.Control` instance connects to a DBus interface, it probably should be created ever only once. If you didn't know already, this class already contains one: https://github.com/elementary/appcenter/blob/0419a054cb4449b72c51cbec4fddfc0a8dc1353a/src/Core/PackageKitBackend.vala#L121.

So instead of creating it everytime when accessing this variable, you should instead have a `private static Pk.Control?` variable that is a singleton:
```
private static Pk.Control? control;
private static unowned Pk.Control get_control () {
    if (control == null) {
        control = new Pk.Control ();
    }

    return control;
}
```

And when it's needed, access it via `get_control ()`.
`supports_parallel_package_queries` could also be static because it doesn't access anything inside the class itself.